frontend/src/app/modules/exclusion-dates/header-date-picker/header-date-picker.component.ts

```diff
@@ -0,0 +1,46 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    Inject,
+} from '@angular/core';
+import { DateAdapter, MAT_DATE_FORMATS, MatDateFormats } from '@angular/material/core';
+import { MatCalendar } from '@angular/material/datepicker';
+import { BaseComponent } from '@core/base/base.component';
+
+@Component({
+    selector: 'app-header-date-picker',
+    styleUrls: ['./header-date-picker.component.sass'],
+    templateUrl: './header-date-picker.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HeaderDatePickerComponent<D> extends BaseComponent {
+    constructor(
+        private calendar: MatCalendar<D>,
+        private dateAdapter: DateAdapter<D>,
+        @Inject(MAT_DATE_FORMATS) private dateFormats: MatDateFormats,
+        cdr: ChangeDetectorRef,
+    ) {
+        super();
+        calendar.stateChanges
+            .pipe(this.untilThis)
+            .subscribe(() => cdr.markForCheck());
+    }
+
+    get periodLabel() {
+        return this.dateAdapter
+            .format(this.calendar.activeDate, this.dateFormats.display.monthYearA11yLabel);
+    }
+
+    previousClicked(mode: 'month' | 'year') {
```

Comment:
Create a type for this

Reply:
I refactored headerdatepicker and there is no need for these params at all